### PR TITLE
Remove Entity<SolutionComponent> from components

### PR DESCRIPTION
**Pull request by 0x6273**
## About the PR
Gets rid of `Entity<SolutionComponent>` from all components. Storing `Entity` in components breaks serialization, and it was redundant because they all store the solution name anyway which is all that's needed to get the solution.

## Breaking changes
`Entity<SolutionComponent>` has been removed from all components. If your code used `ResolveSolution(uid, comp.SolutionName, ref comp.Solution)` you need to change it to `TryGetSolution(uid, comp.SolutionName, out var solutionEnt)`

### CODE REVIEW

**Comment by deltanedas** [comment]
imo Entity should be fixed to serialize properly (just write the uid and when reading resolve the component)

**Comment by 0x6273** [comment]
> imo Entity should be fixed to serialize properly (just write the uid and when reading resolve the component)

Perhaps, but all these components store the solution name anyway, so it's redundant to store a nullable solution entity as well.

**Comment by PJB3005** [comment]
> imo Entity should be fixed to serialize properly (just write the uid and when reading resolve the component)

This isn't happening.

File: Content.Server/Animals/Components/UdderComponent.cs [hunk 1]
```diff
@@ -26,12 +26,6 @@ internal sealed partial class UdderComponent : Component
         [DataField, ViewVariables(VVAccess.ReadOnly)]
         public string SolutionName = "udder";
 
-        /// <summary>
-        ///     The solution to add reagent to.
-        /// </summary>
-        [DataField]
-        public Entity<SolutionComponent>? Solution = null;
```
Comment (ElectroJr) [inline Content.Server/Animals/Components/UdderComponent.cs:33]: The fields themselves are fine, and should stay there to make retrieving solutions faster. you only need to get rid of the DataField attribute

**Review by ElectroJr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**